### PR TITLE
Feat: update border radius variables

### PR DESCRIPTION
**Pull request by marcoambrosini**
@susnux @ShGKme I've been asked to update and tidy this up, could you please tell me if this configuration makes sense to you? Also variable names

Screenshot 2024-07-31 at 17 23 52

### CODE REVIEW

File: apps/theming/lib/Themes/DefaultTheme.php [hunk 1]
```diff
@@ -171,12 +171,22 @@ public function getCSSVariables(): array {
...
+
+			// Border radii (new values)
+			// See url on how to use 
+			'--border-radius-small' => '4px', // For smaller elements
+			'--border-radius-element' => '8px', // For normal elements like buttons and inputs
+			'--border-radius-container' => '12px', // For smaller containers like action menus 
+			'--border-radius-container-large' => '16px', // For bigger containers like body or modals 
+
+			// Border radii (deprecated)
+			'--border-radius' => '4px',
```
Comment (susnux) [inline apps/theming/lib/Themes/DefaultTheme.php:183]:
```suggestion
			'--border-radius' => 'var(--border-radius-small)',
```

File: apps/theming/lib/Themes/DefaultTheme.php [hunk 2]
```diff
@@ -171,12 +171,22 @@ public function getCSSVariables(): array {
...
 			'--border-radius-large' => '10px',
 			'--border-radius-rounded' => '28px',
-			'--border-radius-element' => '8px',
-			// pill-style button, value is large so big buttons also have correct roundness
-			'--border-radius-pill' => '100px',
+			'--border-radius-pill' => '100px', 
+
+			// Big containers border radius, used for the body container
+			'--border-radius-large' => '10px',
```
Comment (susnux) [inline apps/theming/lib/Themes/DefaultTheme.php:189]: Duplicate, already included above.
```suggestion
```

File: apps/theming/lib/Themes/DefaultTheme.php [hunk 3]
```diff
@@ -171,12 +171,22 @@ public function getCSSVariables(): array {
...
+			// Border radii (new values)
+			// See url on how to use 
+			'--border-radius-small' => '4px', // For smaller elements
+			'--border-radius-element' => '8px', // For normal elements like buttons and inputs
+			'--border-radius-container' => '12px', // For smaller containers like action menus 
+			'--border-radius-container-large' => '16px', // For bigger containers like body or modals 
+
+			// Border radii (deprecated)
+			'--border-radius' => '4px',
 			'--border-radius-large' => '10px',
```
Comment (susnux) [inline apps/theming/lib/Themes/DefaultTheme.php:184]:
```suggestion
			'--border-radius-large' => 'var(--border-radius-element)',
```

**Comment by marcoambrosini** [comment]
@susnux @ShGKme  about defult.css and defaulttheme.php, let's decid which file we write documentation in and move it all there?

**Comment by ShGKme** [comment]
> @susnux @ShGKme about defult.css and defaulttheme.php, let's decide which file we write documentation in and move it all there?

Currently, they are documented in the documentation: https://docs.nextcloud.com/server/latest/developer_manual/html_css_design/css.html

**Comment by susnux** [comment]
> @susnux @ShGKme about defult.css and defaulttheme.php, let's decide which file we write documentation in and move it all there?

As @ShGKme mentioned documentation should all go to the developer manual part of the docs, if you ask because of deprecation comments, then I would add those to both files to make it clearer for server developers.

File: apps/theming/lib/Themes/DefaultTheme.php [hunk 4]
```diff
@@ -171,12 +171,22 @@
...
 			'--border-radius-rounded' => '28px',
-			'--border-radius-element' => '8px',
-			// pill-style button, value is large so big buttons also have correct roundness
-			'--border-radius-pill' => '100px',
+			'--border-radius-pill' => '100px', 
+
+			// Big containers border radius, used for the body container
+			'--border-radius-large' => '10px',
```
[the remark this answers is not in the document]
Comment (marcoambrosini) [reply apps/theming/lib/Themes/DefaultTheme.php]: addressed

**Comment by marcoambrosini** [comment]
I added some docs to both files too for now because that's where most people are going to look

**Review by susnux — APPROVED** [review]

**Comment by susnux** [comment]
Looks good, please ensure that you add this to the app upgrade guide in the developer documentation as soon as possible so that app developers can adjust their apps best before Nextcloud 30 release :)

**Review by susnux — APPROVED** [review]

**Comment by marcoambrosini** [comment]
@susnux I also updated the value of body container radius, that ok?

**Review by Antreesy — APPROVED** [review]
Legit

**Comment by skjnldsv** [comment]
Docs in https://github.com/nextcloud/documentation/commit/9a87818f260150f3d2364272fea737cc944b2f6b

**Pull request merged**